controllers/kubeadmconfig_controller.go

```diff
@@ -166,13 +131,83 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
 	}
 
+	// Store Config's state, pre-modifications, to allow patching
+	patchConfig := client.MergeFrom(config.DeepCopy())
+
+	// Check for control plane ready. If it's not ready then we will requeue the machine until it is.
+	// The infrastructure provider *must* set this value to use this bootstrap provider.
```

Comment:
This is set by cluster-api, not an infra provider. See https://github.com/kubernetes-sigs/cluster-api/blob/62582da077611649856cb2a88ec73c957e8569f5/pkg/controller/machine/machine_controller_phases.go#L264-L288

Reply:
this is my comment, yeah, it was set in v1alpha1 by the infra, glad to see it moved to capi

Reply:
fixed